gateway/src/resty/concurrent/timer_task.lua

```diff
@@ -26,15 +25,15 @@ local function generate_id()
   return uuid.generate_v4()
 end
 
-local function mt(id)
-  -- Using 'newproxy' is needed to be able to define __gc().
-  -- With __gc we can make sure that the task will stop scheduling more work
-  -- after it has been garbage collected.
-  local proxy = newproxy(true)
-  local res_mt = getmetatable(proxy)
-  res_mt.__gc = function() _M.unregister_task(id) end
-  res_mt.__index = _M
-  return res_mt
+local function gc(self)
+  _M.unregister_task(self.id)
+end
+
+local function mt()
```

Comment:
Now there is no need to allocate a metatable per instance and can just share one, right?

Reply:
:+1: